### PR TITLE
Clarify optional features available for each protocol

### PR DESCRIPTION
**Pull request by squarooticus**
WIP addressing #17. I don't want to pull this as-is: I just want comments on the format and usefulness of this.

This closes the circle created by the list of protocol features in section 3, and indicates where we probably want more consistency and completeness in that list so the derivation of common optional features is more obvious.

### CODE REVIEW

**Review by chris-wood — COMMENTED** [review]
I like it! It's succinct, and a good way to characterize the optional features across protocols.

**Comment by squarooticus** [comment]
Still a WIP. I have a bunch of comments as a result of doing this exercise:

* Feature availability for some protocol/feature combinations is not clear from the descriptions in the doc.
* Wireguard is IP encapsulation within UDP, but this is not stated anywhere in the doc.
* The set of features we list here should be called out or evident from the protocol features subsections under each protocol description.
* I think there's still some confusion in sections around symmetric authentication vs. signatures, i.e., AEAD-style authentication ("packets belong to this session") vs. PKI endpoint authentication ("this connection belongs to a peer I trust and is not being MITMed"). This may be a comment on the unfortunateness of the literature using "authenticated" for both.

**Comment by chris-wood** [comment]
>Feature availability for some protocol/feature combinations is not clear from the descriptions in the doc.

Can you give examples?

> Wireguard is IP encapsulation within UDP, but this is not stated anywhere in the doc.

Can you please open an issue to address this?

> The set of features we list here should be called out or evident from the protocol features subsections under each protocol description.

See the other open pull request. :-)

> I think there's still some confusion in sections around symmetric authentication vs. signatures, i.e., AEAD-style authentication ("packets belong to this session") vs. PKI endpoint authentication ("this connection belongs to a peer I trust and is not being MITMed"). This may be a comment on the unfortunateness of the literature using "authenticated" for both.

Where in particular are these things confused?

**Comment by chris-wood** [comment]
@squarooticus ping!

**Comment by squarooticus** [comment]
Sorry, am away for the holiday weekend.

>> Feature availability for some protocol/feature combinations is not clear from the descriptions in the doc.

> Can you give examples?

The ones I have listed with a "?", for instance. I guessed on a few of the others, as well (e.g., configuration extensions for MinimalT). The right answer is for me to go back through this table and enumerate all the areas where the text isn't clear, and fix them. I'll file an issue.

>> Wireguard is IP encapsulation within UDP, but this is not stated anywhere in the doc.

> Can you please open an issue to address this?

Will do.

>> I think there's still some confusion in sections around symmetric authentication vs. signatures, i.e., AEAD-style authentication ("packets belong to this session") vs. PKI endpoint authentication ("this connection belongs to a peer I trust and is not being MITMed"). This may be a comment on the unfortunateness of the literature using "authenticated" for both.

> Where in particular are these things confused?

I'm not quite sure what I was thinking here. After reading it through again, I think I'm okay with how this is presented.

**Comment by chris-wood** [comment]
@squarooticus can you please update this PR?

File: draft-ietf-taps-transport-security.md [hunk 1]
```diff
@@ -796,16 +796,16 @@ The following table lists the availability of the above-listed optional features
 |---
 | Protocol | MA | CM | SV | AFN | CX | SC |
```
Comment (csperkins) [inline draft-ietf-taps-transport-security.md]: The last column should be SM, to match the above? ("Session caching and management (SM)")

File: draft-ietf-taps-transport-security.md [hunk 2]
```diff
@@ -796,16 +796,16 @@ The following table lists the availability of the above-listed optional features
...
 |:---------|:--:|:--:|:--:|:---:|:--:|:--:|
-| TLS      | S  | N\* | M | S   | S  | S
+| TLS      | S  | U\* | M | S   | S  | S
 | DTLS     | S  | S  | M  | S   | S  | S
 | IETF QUIC | S | S  | M  | S   | S  | S
 | IKEv2+ESP | M | S  | M  | ?   | S  | S
 | SRTP+DTLS | S | ?  | M  | S   | S  | S
-| SRTP+ZRTP | U | ?  | ?  | U   | U  | N
```
Comment (csperkins) [inline draft-ietf-taps-transport-security.md]: For SRTP, session caching depends on DTLS. I didn't follow DTLS 1.3 development, but if the 0-RTT extensions from TLS 1.3 carried over, then they could presumably be used with SRTP.

File: draft-ietf-taps-transport-security.md (hunk 1 above)
Comment (squarooticus) [reply draft-ietf-taps-transport-security.md]: I think SC makes more sense, but really it's just an abbreviation to make the table representable in ASCII: it doesn't really matter that much.

File: draft-ietf-taps-transport-security.md (hunk 2 above)
Comment (squarooticus) [reply draft-ietf-taps-transport-security.md]: This line is for the SRTP that doesn't employ DTLS for handshake. It's unclear from the text whether it supports session caching, config extensions, or application-layer feature negotiation, so I guessed "no". @chris-wood do you know? Or given the structure with ZRTP as variant, should we even cover it in the table?

Comment (squarooticus) [reply draft-ietf-taps-transport-security.md]: I've reverted them to ? in my latest edit, which I'll push shortly.

**Comment by squarooticus** [comment]
A few more questions:

* Is there any existing client puzzle/proof of work standardized for TLS or DTLS, or is https://tools.ietf.org/html/draft-nygren-tls-client-puzzles-00 as far as something's gotten? Same goes for IPsec; I can find no standardized version of any of these things.
* Should algorithms like TLS, DTLS, etc. that support native authentication but also can export a cryptographic channel binding be listed as supporting authentication delegation? My instinct is yes.

**Comment by chris-wood** [comment]
@squarooticus 

> Is there any existing client puzzle/proof of work standardized for TLS or DTLS, or is https://tools.ietf.org/html/draft-nygren-tls-client-puzzles-00 as far as something's gotten? Same goes for IPsec; I can find no standardized version of any of these things.

None that I'm aware of!

> Should algorithms like TLS, DTLS, etc. that support native authentication but also can export a cryptographic channel binding be listed as supporting authentication delegation? My instinct is yes.

Yes, I think that's correct.

**Comment by chris-wood** [comment]
@squarooticus Will you be able to complete this PR this week?

**Comment by squarooticus** [comment]
Yes, though I could use some guidance on the ?'s in the table, if you have enough context to resolve them. Otherwise, I will do some digging.

**Comment by chris-wood** [comment]
@squarooticus 

IKEv2+ESP/AD: Yes, via EAP(-TLS).
IKEv2+ESP/AFN: Yes, via vendor and configuration payloads. 
SRTP+DTLS/CM: No.
ZRTP: Omit, since it's a variant?
WireGuard/CM: No, connections are bound to IP addresses.
WireGuard/SC: No, session resumption is not supported.
WireGuard/LHP: Yes, for transport packets only.
MinimalT/AD: No.
MinimalT/AFN: No.
MinimalT/LHP: Yes.
CurveCP/AD: No.

**Comment by squarooticus** [comment]
RFR.

**Review by chris-wood — APPROVED** [review]

**Comment by chris-wood** [comment]
Ship it.

**Pull request merged**